src/app/games/teeko/TeekoRules.ts

```diff
@@ -98,32 +99,33 @@ export class TeekoRules extends Rules<TeekoMove, TeekoState> {
         const victoriousCoords: Coord[] = [];
         let zeroSquarePossibilities: number = 0;
         let oneSquarePossibilities: number = 0;
-        for (let cx: number = 0; cx < 3; cx++) {
-            for (let cy: number = 0; cy < 3; cy++) {
+        for (let cx: number = 0; cx < 4; cx++) {
+            for (let cy: number = 0; cy < 4; cy++) {
                 const upLeft: Coord = new Coord(cx, cy);
                 const upRight: Coord = new Coord(cx + 1, cy);
                 const downLeft: Coord = new Coord(cx, cy + 1);
                 const downRight: Coord = new Coord(cx + 1, cy + 1);
-                const upLeftPiece: PlayerOrNone = state.getPieceAt(upLeft);
-                const upRightPiece: PlayerOrNone = state.getPieceAt(upRight);
-                const downLeftPiece: PlayerOrNone = state.getPieceAt(downLeft);
-                const downRightPiece: PlayerOrNone = state.getPieceAt(downRight);
-                const pieces: PlayerOrNone[] = [upLeftPiece, upRightPiece, downLeftPiece, downRightPiece];
+                const pieces: PlayerOrNone[] = [
+                    state.getPieceAt(upLeft),
+                    state.getPieceAt(upRight),
+                    state.getPieceAt(downLeft),
+                    state.getPieceAt(downRight),
+                ];
                 const neutralCount: number = this.count(pieces, PlayerOrNone.NONE);
                 const zeroCount: number = this.count(pieces, PlayerOrNone.ZERO);
                 const oneCount: number = this.count(pieces, PlayerOrNone.ONE);
-                if (neutralCount === 4) {
-                    continue;
+                if (neutralCount === 4) { // TODO: recommencer un partie terminée en phase 2 ne cache pas le last move !!
+                    continue; // Cause four empty space brings nothing to no one
                 } else if (zeroCount + neutralCount === 4) {
-                    if (neutralCount === 0) {
+                    if (neutralCount === 0) { // this square is full of Player.ZERO, victory !
                         victoriousCoords.push(upLeft, upRight, downLeft, downRight);
                     }
-                    zeroSquarePossibilities++;
+                    zeroSquarePossibilities++; // +1 square only occupied by Player.ZERO
                 } else if (oneCount + neutralCount === 4) {
-                    if (neutralCount === 0) {
+                    if (neutralCount === 0) { // this square is full of Player.ONE, victory !
                         victoriousCoords.push(upLeft, upRight, downLeft, downRight);
                     }
-                    oneSquarePossibilities++;
+                    oneSquarePossibilities++; // +1 square only occupied by Player.ONE
```

Comment:
à quoi servent ces variables zeroSquarePossibilities et oneSquarePossibilities ?
Aussi, quand t'as du code qui fait un truc pour chaque joueur, faut préférer for (player in Players.ALL) plutôt que du dupliquer tout !

Reply:
On y gagne précisément zéro ligne (par rapport à ce qu'on a maintenant, pas hier) et ce pour calculer trois fois plus